### ISSUE TITLE
Ninja appears to be downloading stuff during its build

### ISSUE THREAD

**Issue by markmc** [markmc/rebuilding-the-wheel#51]
```
  [ 11%] Performing download step (download, verify and extract) for 'download_ninja_source'
  cd /home/markmc/ai-ml/distro/rebuilding-the-wheel/work-dir/ninja-1.11.1.1/ninja-1.11.1.1 && /usr/bin/cmake -P /home/markmc/ai-ml/distro/rebuilding-the-wheel/work-dir/ninja-1.11.1.1/ninja-1.11.1.1/_skbuild/linux-x86_64-3.9/cmake-build/download_ninja_source-prefix/src/download_ninja_source-stamp/download-download_ninja_source.cmake
  -- Downloading...
     dst='/home/markmc/ai-ml/distro/rebuilding-the-wheel/work-dir/ninja-1.11.1.1/ninja-1.11.1.1/_skbuild/linux-x86_64-3.9/cmake-build/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
     timeout='none'
     inactivity timeout='none'
  -- Using src='https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
  -- Retrying...
  -- Using src='https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
  -- Retry after 5 seconds (attempt #2) ...
  -- Using src='https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
  -- Retry after 5 seconds (attempt #3) ...
  -- Using src='https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
  -- Retry after 15 seconds (attempt #4) ...
  -- Using src='https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
  -- Retry after 60 seconds (attempt #5) ...
  -- Using src='https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz'
  CMake Error at _skbuild/linux-x86_64-3.9/cmake-build/download_ninja_source-prefix/src/download_ninja_source-stamp/download-download_ninja_source.cmake:170 (message):
    Each download failed!

      error: downloading 'https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz' failed
            status_code: 6
            status_string: "Couldn't resolve host name"
            log:
            --- LOG BEGIN ---
            Could not resolve host: github.com
```

**Comment by dhellmann** [comment]
There are some bootstrap build instructions at https://github.com/ninja-build/ninja/wiki that may help.

**Comment by markmc** [comment]
Right, but do we want to build ninja at all? There's an RPM!

I've added a hack to make the wheel wrap the binary from the RPM - see commit 876e1a5e

**Comment by markmc** [comment]
ninja patch not quite there:

```
     Traceback (most recent call last):
      File "/tmp/pip-build-env-lqaahaw0/overlay/bin/ninja", line 5, in <module>
        from ninja import ninja
      File "/tmp/pip-build-env-lqaahaw0/overlay/lib/python3.9/site-packages/ninja/__init__.py", line 30, in <module>
        from skbuild.constants import CMAKE_INSTALL_DIR as SKBUILD_CMAKE_INSTALL_DIR

    ModuleNotFoundError: No module named 'skbuild'
```

**Comment by markmc** [comment]
Resolved with commit 56468c1
